tutorials/tutorial9/tutorial.py

```diff
@@ -0,0 +1,106 @@
+"""Tutorial of the Averaging Neural Operator."""
+
+from time import time
+import torch
+from pina.model import AVNO
+from pina import Condition, LabelTensor
+from pina.problem import AbstractProblem
+from pina.solvers import SupervisedSolver
+from pina.trainer import Trainer
+
+#Data generation
+
+torch.manual_seed(0)
+
+
+def sample_unit_circle(num_points):
+    radius = torch.rand(num_points, 1)
+    angle = torch.rand(num_points, 1) * 2 * torch.pi
+    x = radius * torch.cos(angle)
+    y = radius * torch.sin(angle)
+    data = torch.cat((x, y), dim=1)
+    return data
+
+
+#sin(a*x+b*y)
+def compute_input(data, theta):
+    data = data.reshape(1, -1, 2)
+    z = torch.sin(theta[:, :, 0] * data[:, :, 0] +
+                  theta[:, :, 1] * data[:, :, 1])
+    return z
+
+
+#1+convolution of sin(a*x+b*y) with sin(x) over [0,2pi]x[0,2pi]
+def compute_output(data, theta):
+    data = data.reshape(1, -1, 2)
+    s = torch.cos(theta[:, :, 0] * (torch.pi * data[:, :, 0]) + theta[:, :, 1] *
+                  (torch.pi * data[:, :, 1]))
+    z = 1 - 4 * torch.sin(torch.pi * theta[:, :, 0]) * torch.sin(
+        torch.pi * theta[:, :, 1]) * s / (
+            (theta[:, :, 0]**2 - 1) * theta[:, :, 1])
+    return z
+
+
+theta_dataset = 1 + 0.01 * torch.rand(300, 1, 2)
+data_coarse = sample_unit_circle(1000)
+output_coarse = compute_output(data_coarse, theta_dataset).unsqueeze(-1)
+input_coarse = compute_input(data_coarse, theta_dataset).unsqueeze(-1)
+data_dense = sample_unit_circle(1000)
+output_dense = compute_output(data_dense, theta_dataset).unsqueeze(-1)
+input_dense = compute_input(data_dense, theta_dataset).unsqueeze(-1)
+
+model = AVNO(1, 1, data_coarse, inner_size=500, n_layers=4)
+
```

Comment:
I don't understand. Why can't we pass a tensor 2D tensor?

Reply:
You mean why we can't pass the points directly on the forward pass every time? Theoretically speaking we can, but I don't know if it is compatible with the SupervisedSolver. Alternatively, we can directly concatenate the points with the function before calling the network and pass everything together, but this may confuse the user.

Reply:
Yes I agree, it is tricky. We should find a data structure that works for Neural Operators. Maybe something like we did with DeepOnet? Where in the init of GNO layer we pass something like 'coordinate_labels' for telling which of the labels are coordinates? I think it is also more portable in the long run

Reply:
I changed the structure, so now is more similar to DeepOnet.
.